rfcs/011-dedicated-renku-git-user/011-dedicated-renku-git-user.md

```diff
@@ -0,0 +1,113 @@
+
+# Dedicated Gitlab User for Knowledge Graph
+
+## Summary
+
+Every Renku deployment should have a dedicated regular user in every Gitlab it is connected
+with. This user will be called something like `Renku` or `Renkubot` and it will be used
+by the KG to access the project and process it's history.
+
+## Motivation
+
+The KG needs authenticated access to Gitlab so that it can analyze and index a project's 
+Git history. Currently the KG has to store many Gitlab user tokens in postgres. 
+These tokens are used depending on what project the KG is processing events for. This means
+that the KG has to maintain and manage a database with Gitlab acces tokens (or project tokens).
+If these are lost then the KG would not be able to access this project to add it in its database
+during reprovisioning. 
+
+In addition with the switch from Gitlab user access tokens to project tokens there was a
+lot of automatically generated "bot" users that Gitlab created when the project tokens
+were created. This is just what happens when a Gitlab project token is created but it may
+be cause concerns with some admins or users.
+
+Lastly, with having a regular user being dedicated to the KG, there is no more impersonation
+of users and "fake" user activity that comes from the periodic checks and indexing of projects
+that is done by the KG. All this activity will now be associated with the Renkubot user rather
+than individual real users.
+
+## Design Detail
+
+### Helm Chart
+
+In the Helm chart we would add a hook that would create a user in the Gitlab deployment
+that Renku is connected to. The username and password will be saved in a secret. The KG
+would then have access to the secret and use it to authenticate with Gitlab.
+
+Alternatively the Helm chart can also accept existing user credentials. 
+
+### Gateway
+
+Either as part of this RFC or as an improvement the management of this secret / tokens for 
+the Renkubot user should be done by the Gateway and the KG will simply either ask the Gateway 
+for credentials or a Gateway's forward proxy will inject the right credentials in the right calls.
+
+This will also be helpful when we support multiple Gitlabs/Github. In this case the KG does
+not need to keep track which credentials to use for which Gitlab. The gateway can figure this
+out and inject the right credentials for the Renkubot user.
+
+### UI
+
+There are some changes needed in the UI for this. Namely:
+- To get a list of all projects for a user the UI would have to call Gitlab on behalf
+of the user. This is because the KG will now only know about projects to which the Renkubot
+user has been added. In other words, the KG will not know about any projects for which 
+KG integration is not active. It could figure this out for public Gitlab projects
+but it would be impossible to find private projects that are not in the KG.
+- Creating a project on Renku will have to include adding this Renkubot user to the project
+with `Reporter` permissions.
```

Comment:
It might also be that the Renkubot user would be added to the project during the activation call on the KG side. Obviously, it's not a must, but I can see that building a coherent picture and keeping the responsibilities in the proper places. I might be wrong but for me, it's a part of the activation process.